app/src/app/app-shell/components/shell/shell.component.ts

```diff
@@ -0,0 +1,177 @@
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { AppConfigService, SidenavLayoutComponent, UserPreferencesService } from '@alfresco/adf-core';
+import { Component, OnDestroy, OnInit, ViewChild, ViewEncapsulation } from '@angular/core';
+import { NavigationEnd, Router, NavigationStart } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { Subject, Observable } from 'rxjs';
+import { filter, takeUntil, map, withLatestFrom, delay } from 'rxjs/operators';
+import { NodePermissionService } from '@alfresco/aca-shared';
+import { BreakpointObserver } from '@angular/cdk/layout';
+import { AppStore, getCurrentFolder, getFileUploadingDialog, ResetSelectionAction } from '@alfresco/aca-shared/store';
+import { Directionality } from '@angular/cdk/bidi';
+
+@Component({
+  selector: 'app-shell',
+  templateUrl: './shell.component.html',
+  styleUrls: ['./shell.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  host: { class: 'app-layout' }
+})
+export class ShellLayoutComponent implements OnInit, OnDestroy {
+  @ViewChild('layout', { static: true })
+  layout: SidenavLayoutComponent;
+
+  onDestroy$: Subject<boolean> = new Subject<boolean>();
+  isSmallScreen$: Observable<boolean>;
+
+  expandedSidenav: boolean;
+  currentFolderId: string;
+  canUpload = false;
+
+  minimizeSidenav = false;
+  hideSidenav = false;
+  direction: Directionality;
+
+  showFileUploadingDialog: boolean;
+
+  private minimizeConditions: string[] = ['search'];
+  private hideConditions: string[] = ['/preview/'];
+
+  constructor(
+    protected store: Store<AppStore>,
+    private permission: NodePermissionService,
+    private router: Router,
+    private userPreferenceService: UserPreferencesService,
+    private appConfigService: AppConfigService,
+    private breakpointObserver: BreakpointObserver
+  ) {}
+
+  ngOnInit() {
+    this.isSmallScreen$ = this.breakpointObserver.observe(['(max-width: 600px)']).pipe(map((result) => result.matches));
+
+    this.hideSidenav = this.hideConditions.some((el) => this.router.routerState.snapshot.url.includes(el));
```

Comment:
should this conflict with the `data` property that you have introduced?

Reply:
Fixed